tests/std/tests/P0466R5_layout_compatibility_and_pointer_interconvertibility_traits/test.cpp

```diff
@@ -0,0 +1,179 @@
+    {
+        class A {};
+        class B : public A {};
+        class C : public A {
+            int : 0;
+        };
+// Disable warning C4408: anonymous union did not declare any data members
+#pragma warning(push)
+#pragma warning(disable : 4408)
+        class D : public A {
+            union {};
+        };
+#pragma warning(pop)
+        class E : public A {
+            int v1 = 2;
+        };
+
+        class A1 : public A {};
+        class A2 : public A {};
+        class A3 : public A2 {};
+        class A4 : A1, A3 {};
+
+        ASSERT(is_pointer_interconvertible_base_of_v<A, A>);
+        ASSERT(is_pointer_interconvertible_base_of_v<A, B>);
+        ASSERT(is_pointer_interconvertible_base_of_v<A, C>);
+        ASSERT(is_pointer_interconvertible_base_of_v<A, D>);
+        ASSERT(is_pointer_interconvertible_base_of_v<A, E>);
+        ASSERT(is_pointer_interconvertible_base_of_v<A, A1>);
+        ASSERT(is_pointer_interconvertible_base_of_v<A, A2>);
+        ASSERT(is_pointer_interconvertible_base_of_v<A, A3>);
+        ASSERT(is_pointer_interconvertible_base_of_v<A2, A3>);
+
+        ASSERT(!is_pointer_interconvertible_base_of_v<A4, A>);
+        ASSERT(!is_pointer_interconvertible_base_of_v<A1, E>);
+        ASSERT(!is_pointer_interconvertible_base_of_v<A3, E>);
+    }
+
+    // is_corresponding_member tests
+    {
+        struct S1 {
+            int v1;
+            int v2;
+        };
+
+        struct S2 {
+            int v1;
+            int v2;
+        };
+
+        struct S3 {
+            int v1;
+            int v2;
+            int v3;
+        };
+
+        struct S4 {
+            char v1;
+            int v2;
+            int v3;
+        };
+
+        // Same member - same class
+        ASSERT(is_corresponding_member(&S1::v1, &S1::v1));
+        ASSERT(is_corresponding_member(&S1::v2, &S1::v2));
+
+        // Different member - same class
+        ASSERT(!is_corresponding_member(&S1::v1, &S1::v2));
+        ASSERT(!is_corresponding_member(&S1::v2, &S1::v1));
+
+        // Same member - different class (layout compatible)
+        ASSERT(is_corresponding_member(&S1::v1, &S2::v1));
+        ASSERT(is_corresponding_member(&S1::v2, &S2::v2));
+
+        // Same member - different class (layout compatible)
+        ASSERT(is_corresponding_member(&S1::v1, &S3::v1));
+        ASSERT(is_corresponding_member(&S1::v2, &S3::v2));
+
+        // Different member - different class (layout compatible)
+        ASSERT(!is_corresponding_member(&S1::v1, &S2::v2));
+        ASSERT(!is_corresponding_member(&S1::v2, &S2::v1));
+
+        // Same member - different class (not layout compatible)
+        ASSERT(!is_corresponding_member(&S1::v1, &S4::v1));
+        ASSERT(!is_corresponding_member(&S1::v2, &S4::v2));
+
+        // Same member - different class (not layout compatible)
+        ASSERT(!is_corresponding_member(&S3::v1, &S4::v1));
+        ASSERT(!is_corresponding_member(&S3::v2, &S4::v2));
+    }

```

Comment:
This is missing test cases for corresponding members in the common initial sequence of two different standard-layout types that also have non-corresponding members:
```c++
struct X {
  int v1;
  int v2;
  void* v3;
};

struct Y {
  int v1;
  int v2;
  double v3;
};

ASSERT(is_corresponding_member(&X::v1, &Y::v1));
ASSERT(is_corresponding_member(&X::v2, &Y::v2));
ASSERT(!is_corresponding_member(&X::v1, &Y::v2));
ASSERT(!is_corresponding_member(&X::v2, &Y::v1));
ASSERT(!is_corresponding_member(&X::v3, &Y::v3));
```

Reply:
Added missing test. Names the structs S5 and S6 though for consistency